kratos/utilities/pointer_communicator.h

```diff
@@ -89,6 +89,14 @@ class ResultsProxy
             return mNonLocalData[gp];
     }
 
+    TSendType Get(GlobalPointer<TPointerDataType> const& gp)
```

Comment:
shouldn't this be:

```suggestion
    const TSendType Get(GlobalPointer<TPointerDataType> const& gp)
```

?

Reply:
> shouldn't this be:
> 
> ?

This is the result from the functor, I think there is no reason why it should be constant. What do you think @RiccardoRossi @pooyan-dadvand ?

Reply:
I thought it is common to have a const and a non-const version of a fuction, e.g. [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/node.h#L705-L712) (I now realized that I might be talking abt sth else, but then I am not sure if the non-const version is needed)

Lets wait for the reply of @RiccardoRossi and @pooyan-dadvand, I am not sure on this one

Reply:
@philbucher i don t see a reason for which the returned obj should be const. after retrieving you nay modify it...

regarding the other errror i don t understand hat is happening...

Reply:
@RiccardoRossi the return value should not be const. (You are confused by @philbucher suggestion above) The problem is that you only receive non const GP but what happens if GP is constant and Functor accept it. In this case we need the method added by @ddiezrod 

Answering to @philbucher, The TSendType is the result of the function (for example a copy of pressure) which can be or not constant. So the constantness is reflected in TSendType by the Functor